pkg/nodeagent/store/factory/storefactory.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package factory
+
+import (
+	"fmt"
+
+	"centaurusinfra.io/fornax-serverless/pkg/nodeagent/store"
+	"centaurusinfra.io/fornax-serverless/pkg/nodeagent/store/sqlite"
+)
+
+type PodStore struct {
+	store.Store
+}
+
+type SessionStore struct {
+	store.Store
+}
+
+type ContainerStore struct {
+	store.Store
+}
+
+func NewPodSqliteStore(options *sqlite.SQLiteStoreOptions) (*PodStore, error) {
```

Comment:
do we have to embed sqlite in the func name?

Reply:
this is meant to be Sqlite store, we want to use other underlying store tech in future.